pkg/kinflate/tree/builder.go

```diff
@@ -64,30 +68,38 @@ func manifestToManifestData(m *manifest.Manifest) (*ManifestData, error) {
 }
 
 // loadManifestFileFromPath loads a manifest object from file.
-func loadManifestFileFromPath(filename string) (*manifest.Manifest, error) {
-	loader := kutil.ManifestLoader{fs.MakeRealFS()}
-	m, err := loader.Read(filename)
+func (l *Loader) loadManifestFileFromPath(filename string) (*manifest.Manifest, error) {
+	var loader = kutil.ManifestLoader{FS: l.FS}
+	fullFilename, err := loader.MakeValidManifestPath(filename)
+	if err != nil {
+		return nil, err
+	}
+	l.Path = fullFilename
+	m, err := loader.Read(fullFilename)
 	if err != nil {
 		return nil, err
 	}
+	loader.Validate(m)
 	return m, err
 }
 
-func loadManifestDataFromManifestFileAndResources(m *manifest.Manifest) (*ManifestData, error) {
+func (l *Loader) loadManifestDataFromManifestFileAndResources(m *manifest.Manifest) (*ManifestData, error) {
 	mdata := &ManifestData{}
 	var err error
 	mdata.Name = m.Name
 	mdata.NamePrefix = NamePrefixType(m.NamePrefix)
 	mdata.ObjectLabels = m.ObjectLabels
 	mdata.ObjectAnnotations = m.ObjectAnnotations
 
-	res, err := loadKObjectFromPaths(m.Resources)
+	res, err := l.LoadKObjectFromPaths(m.Resources)
 	if err != nil {
-		return nil, err
+		errorMsg := fmt.Sprintf("Resource from Manifest (%s) couldn't be loaded properly.\n%v\n"+
+			"Please check the Resource subsection in (%s).", l.Path, err, l.Path)
```

Comment:
This is using the Path in the error, but only in the error ... :confused:

Reply:
This will be enhanced once we've fleshed out the ManifestError struct.